qiskit/qasm/node/format.py

```diff
@@ -23,10 +23,13 @@ class Format(Node):
     def __init__(self, value):
         """Create the version node."""
         super().__init__("format", None, None)
-        parts = re.match(r"(\w+)\s+(\d+)\.(\d+)", value)
+        parts = re.match(r"(\w+)\s+(\d+)(\.(\d+))?", value)
+        has_minor_version = re.match("\\.", value)
         self.language = parts.group(1)
         self.majorversion = parts.group(2)
-        self.minorversion = parts.group(3)
+        self.minorversion = "0"
+        if has_minor_version:
+            self.minorversion = parts.group(3)
```

Comment:
You probably meant to match group 4 here (group 3 will include the dot).  You can also avoid the extra boolean value:
```python
self.minorversion = parts.group(4) if parts.group(4) is not None else "0"
```
would be fine as `Match.group` returns `None` for a valid group that isn't matched.  Your current `has_minor_version` is actually wrong; `re.match` searches only for matches at the start of the string, and since it's only valid to be here if the string starts with `OPENQASM`, `has_minor_version` can't be true.

(Entirely up to you, but if you don't like the group numbers, you can also name groups in `re` objects.  You can do 
```python
parts = re.match(r"OPENQASM\s+(?P<major>\d+)(\.(?P<minor>\d+))?", value)
```
and then it's valid to access `parts.group("major")` and `parts.group("minor")`).

Reply:
🤦🏻‍♂️  of course, good catch and thank you.